maven-core/src/main/java/org/apache/maven/toolchain/java/JavaToolchainFactory.java

```diff
@@ -86,9 +86,9 @@ public ToolchainPrivate createToolchain(ToolchainModel model) throws Misconfigur
             throw new MisconfiguredToolchainException(
                     "Java toolchain without the " + JavaToolchainImpl.KEY_JAVAHOME + " configuration element.");
         }
-        File normal = new File(FileUtils.normalize(javahome.getValue()));
+        File normal = Paths.get(javahome.getValue()).normalize().toFile();
```

Comment:
No need to convert to a File; just use Files.exists(path) below

Reply:
Fixed